src/spikeinterface/preprocessing/tests/test_common_reference.py

```diff
@@ -71,6 +71,17 @@ def test_common_reference_channel_slicing(recording):
     local_trace = recording_local_car.get_traces(channel_ids=channel_ids)
 
 
+def test_common_reference_select_channels_local(recording):
+
+    recording_cmr = common_reference(recording, reference="local")
+    recording_segment = recording_cmr._recording_segments[0]
+
+    traces_all = recording_segment.get_traces(start_frame=0, end_frame=10, channel_indices=[0, 1, 2, 3])
+    traces_sub = recording_segment.get_traces(start_frame=0, end_frame=10, channel_indices=[1, 3])
+
+    assert np.all(traces_all[:, [1, 3]] == traces_sub)
```

Comment:
@chrishalcrow instead of this "segment" test, I guess you can trigger the same issue by adding this line:
```
local_trace = recording_local_car.get_traces(channel_ids=["c", "d"])
```
after L71

I would actually prefer to extend that test instead of adding this new one, since it was designed (badly) to test the slicing behavior. Could you do that?

Reply:
I have extended the test to include what you say, and also slicing the segments for each type of common reference. Also made the tests a little harder by choosing ["b", "d"] rather than ["a", "b"]. I also shorted the generated recording from `5s` to `1s`: I can't see any reason here _not_ to do this and it doubles the speed of the test for me locally.